pulser/tests/test_simulation.py

```diff
@@ -307,3 +334,57 @@ def test_run():
                                    np.array([0, sim._times[-10],
                                              sim._times[-3], sim._times[-1]])
                                    )
+
+
+def test_get_xy_hamiltonian():
+    simple_reg = Register.from_coordinates(
+        [[0, 10], [10, 0], [0, 0]], prefix='atom')
+    detun = 1.
+    amp = 3.
+    rise = Pulse.ConstantPulse(1500, amp, detun, 0.)
+    simple_seq = Sequence(simple_reg, MockDevice)
+    simple_seq.declare_channel('ch0', 'mw_global')
+    simple_seq.set_magnetic_field(0, 1., 0.)
+    simple_seq.add(rise, 'ch0')
+
+    assert np.abs(np.linalg.norm(simple_seq.magnetic_field[0:2]) - 1) < 1e-10
+
+    simple_sim = Simulation(simple_seq, sampling_rate=0.01)
+    with pytest.raises(ValueError,
+                       match='less than or equal to the sequence duration'):
+        simple_sim.get_hamiltonian(1650)
+    with pytest.raises(ValueError, match='greater than or equal to 0'):
+        simple_sim.get_hamiltonian(-10)
+    # Constant detuning, so |ud><du| term is C_3/r^3 - 2*detuning for any time
+    simple_ham = simple_sim.get_hamiltonian(143)
+    assert (simple_ham[1, 2] == .5 * MockDevice.interaction_coeff_xy / 10**3)
+    assert np.abs(
+        simple_ham[1, 4] - (-2 * .5 * MockDevice.interaction_coeff_xy / 10**3)
+        ) < 1e-10
+    assert (simple_ham[0, 1] == .5 * amp)
+    assert (simple_ham[3, 3] == -2 * detun)
+
+
+def test_run_xy():
+    simple_reg = Register.from_coordinates(
+        [[10, 0], [0, 0]], prefix='atom')
+    detun = 1.
+    amp = 3.
+    rise = Pulse.ConstantPulse(1500, amp, detun, 0.)
+    simple_seq = Sequence(simple_reg, MockDevice)
+    simple_seq.declare_channel('ch0', 'mw_global')
+    simple_seq.add(rise, 'ch0')
+
+    sim = Simulation(simple_seq, sampling_rate=0.01)
+
+    good_initial_array = np.r_[1, np.zeros(sim.dim**sim._size - 1)]
+    good_initial_qobj = qutip.tensor([qutip.basis(sim.dim, 0)
+                                      for _ in range(sim._size)])
+
+    sim.run(initial_state=good_initial_array)
+    sim.run(initial_state=good_initial_qobj)
+
+    assert not hasattr(sim._seq, '_measurement')
+    simple_seq.measure(basis='XY')
+    sim.run()
+    assert sim._seq._measurement == 'XY'
```

Comment:
Why are you testing this here?

Reply:
Where should it be tested ?

Reply:
This should be tested in `test_sequence.py`, as it pertains to behaviour of `Sequence`, not `Simumation`. In fact, it is already tested there. I was just wondering why you felt the need to do it here too